packages/analytics-browser/test/diagnostic.test.ts

```diff
@@ -0,0 +1,15 @@
+import { Diagnostic } from '../src/diagnostics/diagnostic';
+
+describe('Diagnostic', () => {
+  test('should fetch', async () => {
+    const diagnostic = new Diagnostic();
+    const fetchMock = jest.fn().mockResolvedValueOnce({} as Response);
+    global.fetch = fetchMock;
+
+    diagnostic.track(5, 200, 'Test message');
+    await diagnostic.flush();
+
+    expect(fetchMock).toHaveBeenCalledTimes(1);
```

Comment:
Can you also please verify the track payload here?

Reply:
Verify the payload to be something like 

```
{
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
  },
  body: JSON.stringify({
    api_key: '',
    omni_metrics: [{
      metadata_type: DIAGNOSTIC_METADATA_TYPE,
      library: 'amplitude-ts',
      accounting_time_min: 12345,
      response_code: 400,
      trigger: 'test message 0',
      action: 'drop events',
      event_count: 5,
    }],
  }),
}
```